src/App.tsx

```diff
@@ -1,27 +1,45 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const loadHandler = async (loader: () => Promise<Good[]>) => (
+    setGoods(await loader())
+  );
```

Comment:
It's better to move the result of the async function in a variable and use it in the setState function
```suggestion
  const loadHandler = async (loader: () => Promise<Good[]>) => (
    const goodsFromServer = await loader();
    setGoods(goodsFromServer)
  );
```